packages/kbn-i18n/src/core/i18n.ts

```diff
@@ -235,9 +185,16 @@ export async function load(translationsUrl: string) {
     credentials: 'same-origin',
   });
 
-  if (response.status >= 300) {
+  if (response.status >= 400) {
     throw new Error(`Translations request failed with status code: ${response.status}`);
   }
 
-  init(await response.json());
+  const newTranslation = await response.json();
+  if (!newTranslation || !newTranslation.locale || typeof newTranslation.locale !== 'string') {
```

Comment:
NIT can [also](https://github.com/elastic/kibana/pull/179506/files#r1565517227) be simplified.

And not so NIT: Shouldn't this scenario throw (or at least log) an error?
If we're trying to load and something goes wrong better to know when calling `load()` than when calling `translate()` WDYT?

Reply:
Right some logs or at least an intentional way of how to handle request errors should be in place.
I suggest moving the whole `load` function into core since this doesnt make sense to be at the package level. we have `init` already exposed and called after the translations are loaded in any different way (async require, http request, cache, etc)

I opened an issue to tackle this separately since I dont want to change a lot of the current behavior we have in this upgrade PR to keep the surface area of changes to a minimum as much as possible.

https://github.com/elastic/kibana/issues/181589